Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Spire/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.hpp"
+
+using Filter = Spire::FilteredOrderImbalanceIndicatorModel::Filter;
+using namespace Nexus;
+using namespace Spire;
+
+FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
+  std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
+  std::shared_ptr<std::vector<Filter>> filters)
+  : m_source_model(std::move(source_model)),
+    m_filters(std::move(filters)) {}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    FilteredOrderImbalanceIndicatorModel::load(
+    const TimeInterval& interval) {
+  return m_source_model->load(interval).then(
+    [&filters = *m_filters] (const auto& imbalances) {
```

Comment:
You need to make a copy of ```m_filters```.

Reply:
I thought we were trying to avoid copying them? Or if they're supposed to be a shared_ptr, does this not contradict the first change request?

Reply:
You want to avoid copying the ```vector```, but ```m_filter``` is not a ```vector```, it's a ```shared_ptr``` so copying it is cheap.